check-ts-files.ts

```diff
@@ -0,0 +1,488 @@
+// SPDX-FileCopyrightText: 2024 UnionTech Software Technology Co., Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import fs from 'fs';
+import * as YAML from 'js-yaml';
+import { execSync } from 'child_process';
+import * as Transifex from './transifex';
+import { createTsFileFromTemplate } from './qtlinguist';
+
+// 定义小语种列表
+const MINOR_LANGUAGES = {
+    'es': '西班牙语',
+    'it': '意大利语',
+    'de': '德语',
+    'de_DE': '德语',
+    'ja': '日语',
+    'uk': '乌克兰语',
+    'pt_BR': '巴西葡萄牙语',
+    'sq': '阿尔巴尼亚语'
+};
+
+/**
+ * 从 YAML 文件加载所有项目资源
+ */
+async function loadAllResources() {
+    return await Transifex.getAllLinkedResourcesFromProjects(
+        YAML.load(fs.readFileSync('./transifex-projects.yml', 'utf8')) as string[]
+    );
+}
+
+/**
+ * 从 Transifex 同步翻译文件
+ */
+function syncTranslationsFromTransifex(repository: string, repoPath: string) {
+    console.log(`正在从Transifex平台同步${repository}仓库的翻译文件...`);
+    try {
+        // 检查仓库中是否有.tx目录
+        const txConfigPath = `${repoPath}/.tx/config`;
+        const txYamlPath = `${repoPath}/.tx/transifex.yaml`;
+        
+        if (fs.existsSync(txYamlPath) || fs.existsSync(txConfigPath)) {
```

Comment:
```suggestion
        if (fs.existsSync(txConfigPath)) {
```

tx pull 是必须有 .tx/config ，`transifex.yaml` 不能起到相同作用。

如果预期是 transifex.yaml 的也拉取的话，就需要用 deepin-translation-utils 生成 .tx/config 了。

另外有个别项目的 .tx/config 是有问题的，需要在实际启用此脚本前人工检查一下。